### PR TITLE
ENG-12282: accept mastership for non-drained non-continueing generation.

### PR DESCRIPTION
**Pull request by akhanzode**
accept mastership for non-drained non-continueing generation.

### CODE REVIEW

File: src/frontend/org/voltdb/export/ExportDataSource.java [hunk 1]
```diff
@@ -949,9 +954,12 @@ public void run() {
         stashOrSubmitTask(runnable, true, false);
     }
 
-     private void ackImpl(long uso, Exception ackCallStack) {
+     private synchronized void ackImpl(long uso, Exception ackCallStack) {
```
Comment (nshi) [inline src/frontend/org/voltdb/export/ExportDataSource.java]: Is it necessary to synchronize this?

File: src/frontend/org/voltdb/export/ExportDataSource.java [hunk 2]
```diff
@@ -89,6 +89,8 @@
     private final StreamBlockQueue m_committedBuffers;
     private boolean m_endOfStream = false;
     private Runnable m_onDrain;
+    private boolean m_drained = false;
+    private final boolean m_isInContinueingGeneration;
```
Comment (nshi) [inline src/frontend/org/voltdb/export/ExportDataSource.java]: Do we still need this flag?

Comment (akhanzode) [reply src/frontend/org/voltdb/export/ExportDataSource.java]: No we dont.

File: src/frontend/org/voltdb/export/ExportDataSource.java [hunk 3]
```diff
@@ -145,6 +145,7 @@ public ExportDataSource(
             public void run() {
                 try {
                     //Set end of stream so in case we become master we will finish up and close.
+                    exportLog.info("Setting EOS on Drain no push should come");
```
Comment (nshi) [inline src/frontend/org/voltdb/export/ExportDataSource.java]: Do we need this info level message? Looks more like a debug message.

Comment (akhanzode) [reply src/frontend/org/voltdb/export/ExportDataSource.java]: Yes it is...My debugging leftover.

**Pull request merged**